core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -524,6 +529,11 @@ private[spark] class AppStatusStore(
         } else {
           None
         }
+      val speculationStageSummary: Option[v1.SpeculationStageSummary] = if (withSummaries) {
```

Comment:
I didn't quite get it. We are populating `speculationSummary` data only when `withSummaries` is true if not it is `None`. Am I missing something here?

Reply:
How about check the condition of `withDetail` instead of `withSummaries` like `executorSummaries? Then, the new REST API `/speculationSummary` is not needed because `withDetail` is `true` by default when we access to a path of the form of `/stage/<stageId>/<attemptId>`.

Reply:
Done.